deps/CMakeLists.txt

```diff
@@ -86,11 +86,61 @@ if(NOT EMSCRIPTEN)
         libressl
         URL https://develop.fragcolor.com/libressl-3.9.2.tar.gz
         URL_HASH SHA256=7b031dac64a59eb6ee3304f7ffb75dad33ab8c9d279c847f92c89fb846068f97
-        ${LIBRESSL_PATCH_COMMAND}
       )
       FetchContent_MakeAvailable(libressl)
     endif()
 
+    # Fix iOS endian compatibility issue after LibreSSL is available
+    if(APPLE AND (IOS OR CMAKE_OSX_SYSROOT MATCHES ".*iphoneos.*" OR CMAKE_OSX_SYSROOT MATCHES ".*iphonesimulator.*" OR CMAKE_OSX_SYSROOT MATCHES ".*xros.*"))
+      message(STATUS "Fixing iOS endian compatibility in LibreSSL source files...")
+
+      # Define the replacement text for iOS endian compatibility
+      set(IOS_ENDIAN_REPLACEMENT "/* iOS/Apple platforms endian function compatibility */
```

Comment:
This might be cleaner to fix by just patching  `include/compat/endian.h`
Because it has this check but picks up a random endian.h that is the wrong one it seems
https://github.com/libressl/portable/blob/55f21a399f56695990d5495384146b55a5e56823/include/compat/endian.h#L65-L77

https://github.com/libressl/portable/issues/1025